auth/standard_role_manager.cc

```diff
@@ -701,4 +703,150 @@ future<> standard_role_manager::remove_attribute(std::string_view role_name, std
+    };
+
+    for (size_t idx = 0; idx < roles.size(); ++idx) {
+        const bool role_can_login = co_await can_login(roles[idx]);
+        const bool role_is_superuser = co_await is_superuser(roles[idx]);
+        const sstring formatted_role_name = cql3::util::maybe_quote(roles[idx]);
+
+        result.push_back(cql3::description {
+            // Roles do not belong to any keyspace.
+            .keyspace = std::nullopt,
+            .type = "role",
+            .name = formatted_role_name,
+            .create_statement = produce_create_statement(
+                    formatted_role_name,
+                    with_salted_hashes ? maybe_salted_hashes[idx] : std::nullopt,
+                    role_can_login,
+                    role_is_superuser)
+        });
+    }
+
+    co_return result;
 }
+
+future<std::vector<cql3::description>> standard_role_manager::describe_role_grants() const {
+    std::vector<cql3::description> result{};
+
+    const auto grants = co_await query_all_directly_granted();
+    result.reserve(grants.size());
+
+    for (const auto& [grantee_role, granted_role] : grants) {
+        const auto formatted_grantee = cql3::util::maybe_quote(grantee_role);
+        const auto formatted_granted = cql3::util::maybe_quote(granted_role);
+
+        result.push_back(cql3::description {
+            // Role grants do not belong to any keyspace.
+            .keyspace = std::nullopt,
+            .type = "grant_role",
+            .name = formatted_granted,
+            .create_statement = seastar::format("GRANT {} TO {};", formatted_granted, formatted_grantee)
+        });
+
+        co_await coroutine::maybe_yield();
+    }
+
+    co_return result;
+}
+
+future<std::vector<cql3::description>> standard_role_manager::describe_attached_service_levels() const {
+    std::vector<cql3::description> result{};
+
+    // Note: As of now, the only attributes we can use are service levels.
+    //
+    // Note: We keep in mind this fragment of the documentation:
+    //       "A role can only be assigned one service level. However, the same service level
+    //       can be attached to many roles. If a role inherits a service level from another role,
+    //       the highest level of service from all the roles wins."
+    const auto attached_service_levels = co_await query_attribute_for_all("service_level");
+    result.reserve(attached_service_levels.size());
+    
+    for (const auto& [role, service_level] : attached_service_levels) {
+        const auto formatted_role = cql3::util::maybe_quote(role);
+        const auto formatted_sl = cql3::util::maybe_quote(service_level);
+
+        result.push_back(cql3::description {
+            // Attaching a service level doesn't belong to any keyspace.
+            .keyspace = std::nullopt,
+            .type = "service_level_attachment",
+            .name = formatted_sl,
+            .create_statement = seastar::format("ATTACH SERVICE_LEVEL {} TO {};", formatted_sl, formatted_role)
```

Comment:
That _ is going to haunt me forever

Reply:
The underscore will remain in the grammar forever, but we could extend it by an alternative `SERVICE LEVEL | SERVICE_LEVEL` (if it's not a thing yet)

Reply:
yes

Reply:
There's also `WORKLOAD_TYPE` that could get an alternative `WORKLOAD TYPE`